docs/microphone.rst

```diff
@@ -70,11 +119,61 @@ Functions
     * **return**: a representation of the sound pressure level in the range 0 to
       255.
 
+.. py:function:: record(duration=3000, rate=7812, wait=True)
+
+    Record sound for the amount of time indicated by ``duration`` at the
+    sampling rate indicated by ``rate``.
+
+    The amount of memory consumed is directly related to the length of the
+    recording and the sampling rate. The higher these values, the more memory
+    it will use.
+
+    A lower sampling rate will reduce both memory consumption and sound
+    quality.
+
+    If there isn't enough memory available a ``MemoryError`` will be raised.
+
+    :param duration: How much time to record in milliseconds.
+    :param rate: Number of samples to capture per second.
+    :param wait: When set to ``True`` it blocks until the recording is
+        done, if it is set to ``False`` it will run in the background.
+    :returns: An ``AudioFrame`` with the sound samples.
 
-Example
-=======
+.. py:function:: record_into(buffer, rate=7812, wait=True)
```

Comment:
Perhaps this function can set some new internal state in an `AudioFrame` object which indicates the length of the recording (as opposed to the total allocated length of the buffer).  Then `audio.play()` would use this state to only play the amount that was recorded.

Could then add a method to get/set this length, eg `AudioFrame.get_recording_length()`, and `AudioFrame.set_recording_length()`.

Reply:
Yes, I completely agree here, playing with the current branch and recording a couple of seconds into a 5 second "buffer" results in a few seconds of "silent playback".

Would `record_into()` then also use this marker to continue recording where it left off?

Reply:
> Would `record_into()` then also use this marker to continue recording where it left off?

I think that would be confusing.  It would require the user to explicitly reset the marker to the beginning to reuse the buffer.  Maybe instead `record_into()` could have an additional argument which lets the user specify the starting location in the buffer.

Reply:
Discussion:
- https://github.com/microbit-foundation/micropython-microbit-v2/issues/197